### DISCUSSION TITLE
Allow development builds

### DISCUSSION THREAD

**Discussion by Rc85** [vercel/next.js#46471] [Ideas]
### Describe the feature you'd like to request

I would like to have the option to set the `.env` to use when running `next build`. The reason is that I want to build my app on my machine before deploying to my VPS to save the hassle of uploading all the unnecessary files (eg. typescript files) to VPS. On my VPS, I have two environments, staging and production. Building the staging environment app on my machine using `.env.development` will reduce a lot of uploading time and resources.

OS: Windows 10 Pro
NextJS: 12.3.1
Node: 16.15.0

When I set `NODE_ENV` the non standard way, it is ignored.
![image](https://user-images.githubusercontent.com/22307591/221351633-06914635-7099-4a67-983e-983ee290d0cc.png)

![image](https://user-images.githubusercontent.com/22307591/221351664-4beb9c98-b530-443b-9de4-430711df6a1f.png)

### Describe the solution you'd like

Implement an option when running `next build` such as `--env development`.

### Describe alternatives you've considered

Use `NEXT_ENV` to determine which `.env` file to load so it doesn't interfere with `NODE_ENV`.

**Comment by TheMikeyRoss** [comment]
Im currently stuck in this part.

@Rc85 Did you figure out how to setup a custom staging environment?

**Comment by himself65** [comment]
https://github.com/vercel/next.js/blob/canary/test/production/allow-development-build/allow-development-build.test.ts
